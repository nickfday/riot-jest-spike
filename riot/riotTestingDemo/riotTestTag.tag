<riotTestTag><div class="test-wrapper"><h1>Riot Test Header</h1><h3>{opts.title}</h3></div></riotTestTag>