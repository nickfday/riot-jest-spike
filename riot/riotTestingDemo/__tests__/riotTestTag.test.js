import { mount, shallow } from 'riot-test-utils';
import assert from 'assert';
import riotTestTag from '../riotTestTag.tag';

var testTag = riotTestTag;
var wrapper = mount(testTag, {
  title: 'Fake Title'
});

// snapshot test
it('card should match snapshot', function() {
  expect(wrapper.toJSON()).toMatchSnapshot();
});

// DOM testing
it('should display the h1 static title', () => {
  assert(wrapper.root.querySelector('h1').textContent === 'Riot Test Header');
});

it('should display the h3 dynamic title', () => {
  assert(wrapper.root.querySelector('h3').textContent === 'Fake Title');
});

it('should match the expected outerHTHML exactly', () => {
  assert(
    wrapper.root.querySelector('.test-wrapper').outerHTML ===
      '<div class="test-wrapper"><h1>Riot Test Header</h1><h3>Fake Title</h3></div>'
  );
});
