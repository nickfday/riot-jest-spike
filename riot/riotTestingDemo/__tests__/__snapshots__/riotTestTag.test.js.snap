// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`card should match snapshot 1`] = `
<riottesttag
  data-is={
    Array [
      "riottesttag",
    ]
  }
>
  <div
    class={
      Array [
        "test-wrapper",
      ]
    }
  >
    <h1>
      Riot Test Header
    </h1>
    <h3>
      Fake Title
    </h3>
  </div>
</riottesttag>
`;
